tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -38,19 +40,129 @@ def test_slurm(region, pcluster_config_reader, clusters_factory):
+    )
+    result = remote_command_executor.run_remote_command(
+        "sbatch -G {0} --wrap='sleep 1'".format(num_gpus * max_queue_size + 1), raise_on_error=False
+    )
+    assert_that(result.stdout).contains(
+        "sbatch: error: Batch job submission failed: Requested node configuration is not available"
+    )
+
+    # Commands below should be correctly submitted
+    result = remote_command_executor.run_remote_command(
+        "sbatch -N 1 --wrap='sleep 1' -G {0} --gpus-per-task=1 -n {1}".format(num_gpus, num_gpus)
+    )
+    result = remote_command_executor.run_remote_command("sbatch -N 1 --wrap='sleep 1' --gres=gpu:{0}".format(num_gpus))
+    result = remote_command_executor.run_remote_command(
+        "sbatch -G {0} --gpus-per-node={1} --wrap='sleep 1'".format(num_gpus * max_queue_size, num_gpus)
+    )
+
+
+def _gpu_test_conflicting_options(remote_command_executor, num_gpus):
+    """Test GPU-related conflicting option senerios."""
+    logging.info("Testing scheduler does not accept jobs when requesting job containing conflicting options")
+
+    result = remote_command_executor.run_remote_command(
+        "sbatch -G 1 --cpus-per-gpu 32 -N 1 --cpus-per-task 20 --wrap='sleep 1'", raise_on_error=False
+    )
+    assert_that(result.stdout).contains(
+        "sbatch: error: Batch job submission failed: Requested node configuration is not available"
+    )
+
+
+def _gpu_resource_check(remote_command_executor):
+    """Test GPU related resources are correctly allocated."""
+    logging.info("Testing number of GPU/CPU resources allocated to job")
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    result = remote_command_executor.run_remote_command("sbatch -G 1 --cpus-per-gpu 5 --wrap='sleep 1'")
+    job_id = slurm_commands.assert_job_submitted(result.stdout)
+
+    job_info = _get_job_info(remote_command_executor, job_id)
+    assert_that(job_info).contains("TresPerJob=gpu:1", "CpusPerTres=gpu:5")
+
+    result = remote_command_executor.run_remote_command("sbatch --gres=gpu:2 --cpus-per-gpu 6 --wrap='sleep 1'")
+    job_id = slurm_commands.assert_job_submitted(result.stdout)
+
+    job_info = _get_job_info(remote_command_executor, job_id)
+    assert_that(job_info).contains("TresPerNode=gpu:2", "CpusPerTres=gpu:6")
+
+
+def _gpu_test_scaleup(remote_command_executor, region, asg_name, stack_name, scaledown_idletime, num_gpus):
+    """Test cluster is scaling up correctly and GPU jobs are not aborted on slurmctld restart."""
+    logging.info("Testing cluster scales correctly with GPU jobs")
+    slurm_commands = SlurmCommands(remote_command_executor)
+    # Assert initial conditions
+    _assert_asg_has_no_node(region, asg_name)
+    _assert_no_nodes_in_scheduler(slurm_commands)
+
+    # Should require 3 nodes
+    result = remote_command_executor.run_remote_command(
+        "sbatch --wrap 'sleep 10' -c 10 -n 4 -G {0}".format(num_gpus * 2 + 1)
```

Comment:
Add tests for these 3 jobs, submitted at the same time: --gpus=3, --gpus-per-task=1 -n 3, --gpus-per-node 1 -N 1.